### PR TITLE
script: Re-exec release-flynn from commit being released

### LINKED ISSUE #652: Release doesn't use packer scripts from version being released
The nightly releases do not appear to be using the packer scripts from the git tree that is being released. I think we should re-exec the build script after checking out the correct version.

### PR DESCRIPTION
**Pull request by lmars**
Fixes #652

### CODE REVIEW

File: script/release-flynn [hunk 1]
```diff
@@ -68,6 +68,27 @@ main() {
...
+    info "re-executing using scripts from commit ${commit}"
+
+    src="$(mktemp -d)"
+    trap "rm -f ${src}" EXIT
+    git clone --quiet https://github.com/flynn/flynn "${src}"
+    pushd "${src}" >/dev/null
+    git checkout --force --quiet "${commit}"
+    popd >/dev/null
+
+    "${src}/script/release-flynn" \
```
Comment (titanous) [inline script/release-flynn]: Any reason not to `exec` this?

Comment (titanous) [inline script/release-flynn]: Ah, to avoid the trap, nvm.

**Comment by titanous** [comment]
LGTM

File: script/release-flynn [hunk 2]
```diff
@@ -68,6 +68,27 @@ main() {
       fail "could not determine commit"
     fi
   fi
+
+  info "checking script is from commit being released"
+  local src_commit="$(git rev-parse HEAD)"
+  if [[ "${src_commit}" != "${commit}" ]]; then
+    info "re-executing using scripts from commit ${commit}"
+
+    src="$(mktemp -d)"
```
Comment (titanous) [inline script/release-flynn]: `local`?

**Comment by lmars** [comment]
@titanous comments addressed, plus updated to build the `flynn-release` binary

**Comment by titanous** [comment]
LGTM

**Pull request merged**